### DISCUSSION TITLE
Email Magic Link Authentication Not Working in Self-Hosted Outline Wiki

### DISCUSSION THREAD

**Discussion by koryo93** [outline/outline#6155] [Self Hosting]
Hi everyone,

I'm trying to set up email magic link authentication for my self-hosted Outline wiki on a Rocky Linux server. I've configured SMTP settings and confirmed they work using swaks, but when I navigate to the login page, it does not prompt for an email address to send the magic link.

Environment:

Outline version: latest
OS: Rocky Linux 8.8
Database: PostgreSQL 13
SSL: Let's Encrypt with Nginx reverse proxy
Configuration:
# docker-compose.yml snippet
environment:
  ENABLED_SIGNIN_EMAIL: "true"
  SMTP_HOST: "smtp servername"
  SMTP_PORT: "587"
  # ... other settings ...

Steps Taken:

Sent test email with swaks successfully.
Reviewed Outline logs; no errors related to email.
Checked database tables; they are present.
Actual vs. Expected Behavior: The login page doesn't provide an option to enter my email for a magic link. Instead, it shows "Continue with Outline" with no further action available.

![outline](https://github.com/outline/outline/assets/8231739/bd7a6e24-db37-4596-9b63-db01b02cb136)

Logs:
# docker-compose logs outline
/usr/local/lib/python3.6/site-packages/paramiko/transport.py:32: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography. The next release of cryptography will remove support for Python 3.6.
  from cryptography.hazmat.backends import default_backend
Attaching to outline
outline     | yarn run v1.22.19
outline     | $ node ./build/server/index.js
outline     | {"label":"lifecycle","level":"info","message":"Note: Restricting process count to 1 due to use of collaborative service"}
outline     | (node:28) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
outline     |
outline     | Please migrate your code to use AWS SDK for JavaScript (v3).
outline     | For more information, check the migration guide at https://a.co/7PzMCcy
outline     | (Use `node --trace-warnings ...` to show where the warning was created)
outline     | {"label":"lifecycle","level":"info","message":"\nIs your team enjoying Outline? Consider supporting future development by sponsoring the project:\n\nhttps://github.com/sponsors/outline\n"}
outline     | {"label":"lifecycle","level":"info","message":"Note: Restricting process count to 1 due to use of collaborative service"}
outline     | {"label":"lifecycle","level":"info","message":"Starting collaboration service"}
outline     | {"label":"lifecycle","level":"info","message":"Starting websockets service"}
outline     | {"label":"lifecycle","level":"info","message":"Starting worker service"}
outline     | {"label":"lifecycle","level":"info","message":"Starting web service"}
outline     | (node:39) NOTE: We are formalizing our plans to enter AWS SDK for JavaScript (v2) into maintenance mode in 2023.
outline     |
outline     | Please migrate your code to use AWS SDK for JavaScript (v3).
outline     | For more information, check the migration guide at https://a.co/7PzMCcy
outline     | (Use `node --trace-warnings ...` to show where the warning was created)
outline     | {"label":"lifecycle","level":"info","message":"Listening on http://localhost:3000 / https://outline.gomcorp.com"}

Sorry, https://outline.gomcorp.com is blocked by a firewall.
Can anyone assist me in enabling email magic link authentication? Are there additional logs or settings I should look at?

Thanks in advance for any help!

**Comment by tommoor** [comment]
If you don't want to use SSO you'll need to generate the workspace manually, this isn't ideal right now – it must be done with seed script. Something like: 

`docker-compose run --rm outline node build/server/scripts/seed.js my@email.com`

**Reply by JustDoItSascha** [reply]
A thank you, was looking exactly for that. Maybe it should be added to the documentation?

**Reply by ellotheth** [reply]
It's in the self-hosting documentation, but [under the *Local development* heading](https://docs.getoutline.com/s/hosting/doc/local-development-5hEhFRXow7#h-authentication), so it's easy to miss.

**Comment by ovizii** [comment]
I'm sorry to revive this thread but where did you find this variable:

`ENABLED_SIGNIN_EMAIL: "true"`

I cannot find any reference to it other than this thread. :-(

**Reply by ZQlby** [reply]
![1](https://github.com/outline/outline/assets/28595778/6acf6337-4cff-441d-8588-fa6614cebc5f)
still cant open up
